### DISCUSSION TITLE
Returns Estimation: CAPM and Merton

### DISCUSSION THREAD

**Discussion by c1adrien** [skfolio/skfolio#33] [Ideas]
Hi everyone, I wanted to suggest implementing a new method for estimating asset returns. I came across Merton's article (Merton, R. C. (1980). On estimating the expected return on the market: An exploratory investigation. Journal of Financial Economics, 8(4), 323-361.)
The article builds upon the Capital Asset Pricing Model (CAPM) :

$$
\alpha_i-r=\beta_i(\alpha-r)
$$

where $\alpha_i$ and $\alpha$ denotes the expected rate of return on security $i$ and the market portfolio, respectively; $r$ is the riskless interest rate; and $\beta_i$ is the ratio of the covariance of the return on security $i$ with the return on the market divided by variance of the return on the market.

Merton revisits the idea that one must take into account interest rates to estimate returns since an equilibrium condition implies that $\alpha_i>r$. This allows for the consideration of inflation. Additionally, Merton emphasizes that the excess return over the risk-free rate is not necessarily constant over time and that the excess return should be explained by an increasing function $g$ of risk in the market. Risk is measured by the variance of returns, and by introducing a state variable $Y$, the following model is obtained:

$$
\alpha-r=Y g\left(\sigma^2\right)
$$

The state variable Y is used to account for the fact that the risk aversion function may vary over time. 

In his article, Merton proposes three models, specifying the function g each time, and he also suggests estimators for the variable $Y$. I think it could be interesting to implement the latter, which are close to the framework of the CAPM. What do you think?

Thanks !

**Comment by HugoDelatte** [comment]
That's a great idea. `skfolio` is a toolkit, and I think this methods will make it more complete.
You can go ahead with the implementation, a good place to start is to have a look at the following  estimators:
* [EquilibriumMu](https://github.com/skfolio/skfolio/blob/1b0c8811fe1d7f2dc6aef31bf82e9694a59617b5/src/skfolio/moments/expected_returns/_expected_returns.py#L128)
* [ShrunkMu](https://github.com/skfolio/skfolio/blob/1b0c8811fe1d7f2dc6aef31bf82e9694a59617b5/src/skfolio/moments/expected_returns/_expected_returns.py#L238)

It will help you in term of API design :)
